cmd/gitserver/server/cleanup_test.go

```diff
@@ -15,14 +16,74 @@ import (
 	"testing/quick"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
 	"github.com/pkg/errors"
+	"github.com/sourcegraph/sourcegraph/internal/gitserver/protocol"
 )
 
 const (
 	testRepoA = "testrepo-A"
 	testRepoC = "testrepo-C"
 )
 
+func TestCleanup_computeStats(t *testing.T) {
+	root, err := ioutil.TempDir("", "gitserver-test-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(root)
+
+	for _, name := range []string{"a", "b/d", "c"} {
+		p := path.Join(root, name, ".git")
+		if err := os.MkdirAll(p, 0755); err != nil {
+			t.Fatal(err)
+		}
+		cmd := exec.Command("git", "--bare", "init", p)
+		if err := cmd.Run(); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	want := protocol.ReposStats{
+		UpdatedAt: time.Now(),
+
+		// This may be different in practice, but the way we setup the tests
+		// we only have .git dirs to measure so this is correct.
+		GitDirBytes: dirSize(root),
+	}
+
+	// We run cleanupRepos because we want to test as a side-effect it creates
+	// the correct file in the correct place.
+	s := &Server{ReposDir: root}
+	s.Handler() // Handler as a side-effect sets up Server
+	s.cleanupRepos()
+
+	// we hardcode the name here so the tests break if someone changes the
+	// value of reposStatsName. We don't want it to change without good reason
+	// since it will temporarily break the repo-stats endpoint.
+	b, err := ioutil.ReadFile(filepath.Join(root, "repos-stats.json"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var got protocol.ReposStats
+	if err := json.Unmarshal(b, &got); err != nil {
+		t.Fatal(err)
+	}
+
+	if got.UpdatedAt.Before(want.UpdatedAt) {
+		t.Fatal("want should of been computed after we called cleanupRepos")
```

Comment:
nit (french guy here, I might be wrong): 
```suggestion
		t.Fatal("want should have been computed after we called cleanupRepos")
```